src/core/routing/transfer-routes/network-paths.ts

```
@@ -0,0 +1,123 @@
+
+    const queue: { line: string; path: string[]; transfers: number }[] = [];
+    
+    // Keep track of paths reaching each line for pruning duplicates
+    const bestTransfersToLine: Map<string, number> = new Map();
+    bestTransfersToLine.set(originLine, 0);
+    
+    // Start with the origin line
+    queue.push({ line: originLine, path: [originLine], transfers: 0 });
+    
+    while (queue.length > 0) {
+      const { line, path, transfers } = queue.shift()!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)